tests/src/3d/testqgs3dutils.cpp

```diff
@@ -516,6 +519,88 @@ void TestQgs3DUtils::testScreenPointToMapCoordinates()
   QGSCOMPARENEAR( mapPoint.z(), -252, 2 );
 }
 
+void TestQgs3DUtils::testLineSegmentToClippingPlanes()
+{
+  const QgsPointXY point1( 20, 20 );
+  const QgsPointXY point2( 50, 50 );
+  const QgsVector3D midPoint( point2.x() - point1.x(), point2.y() - point1.y(), 0 );
+  QList<QVector4D> clippingPlanes = Qgs3DUtils::lineSegmentToClippingPlanes( point1, point2, 10, QgsVector3D( 0, 0, 0 ) );
+  for ( int i = 0; i < clippingPlanes.size(); i++ )
+  {
+    QgsVector3D planePoint(
+      i % 2 ? midPoint.x() : ( -clippingPlanes.at( i ).y() * midPoint.y() - clippingPlanes.at( i ).w() ) / clippingPlanes.at( i ).x(),
+      i % 2 ? ( -clippingPlanes.at( i ).x() * midPoint.x() - clippingPlanes.at( i ).w() ) / clippingPlanes.at( i ).y() : midPoint.y(),
+      0
+    );
+    const double distance = QgsVector3D::dotProduct( midPoint - planePoint, clippingPlanes.at( i ).toVector3D() );
+    // verify all normals are pointing inside the rectangle
+    QVERIFY( distance > 0 );
+  }
```

Comment:
let's check we get exactly 4 clipping planes as a result...

and let's simplify this - pick just one point that's clearly inside (e.g. 30, 30) and check that it is inside (i.e. all dot products are positive).

and let's also take a point that's outside and check that at least one dot product is negative...